### PR TITLE
Add docs for respond_to

### PR DESCRIPTION
**Pull request by jez**
### Test plan

See included automated tests.

### CODE REVIEW

File: website/docs/flow-sensitive.md [hunk 1]
```diff
@@ -96,6 +96,63 @@ The complete list of constructs that affect Sorbet's flow-sensitive typing:
...
+  if x.respond_to?(:foo)
+    T.reveal_type(x) # => Object
+    x.foo # Method `foo` does not exist
+  end
+end
+```
+
+In this example, knowing that `x` responds to a method with the name `foo` does
+not tell Sorbet anything more specific about the type for `x`. Sorbet does not
+have any sort of duck-typed interfaces that let Sorbet update it's knowledge of
```
Comment (froydnj) [inline website/docs/flow-sensitive.md:119]:
```suggestion
have any sort of duck-typed interfaces that let Sorbet update its knowledge of
```

File: website/docs/flow-sensitive.md [hunk 2]
```diff
@@ -96,6 +96,63 @@ The complete list of constructs that affect Sorbet's flow-sensitive typing:
...
+the type of `x` to, "`Object` plus responds to `foo`" so it must keep the type
```
Comment (froydnj) [inline website/docs/flow-sensitive.md:120]:
```suggestion
the type of `x` to "`Object` plus responds to `foo`", so it must keep the type
```

Not 100% sure that the added comma is really necessary.

File: website/docs/flow-sensitive.md [hunk 3]
```diff
@@ -96,6 +96,63 @@ The complete list of constructs that affect Sorbet's flow-sensitive typing:
...
+of `x` at `Object`, which does not allow calling `foo`.
+
+Note that even if Sorbet did support such a type, it's likely that `respond_to?`
+would _still_ not be supported, because knowing that the method `foo` exists
```
Comment (froydnj) [inline website/docs/flow-sensitive.md:124]: Is "still not be supported" clearly referring to support for flow sensitivity in this context?  I can see the argument for, but it still feels slightly jarring to read.

**Review by froydnj — APPROVED** [review]
Nice, small suggestions only.

File: website/docs/flow-sensitive.md (hunk 3 above)
Comment (jez) [reply website/docs/flow-sensitive.md]: clarified

File: website/docs/flow-sensitive.md [hunk 4]
```diff
@@ -96,6 +96,64 @@ The complete list of constructs that affect Sorbet's flow-sensitive typing:
...
+the type of `x` to "`Object` plus responds to `foo`", so it must keep the type
+of `x` at `Object`, which does not allow calling `foo`.
+
+Note that even if Sorbet did support such a type, it's likely that
+flow-sensitive type updates for `respond_to?` would _still_ not be supported,
+because knowing that the method `foo` exists says nothing about what parameters
+that method expects, what their types are, or what the return type of that
+function is.
+
+It's possibly someday that Sorbet could support a limited form of
```
Comment (BrianHicks) [inline website/docs/flow-sensitive.md:129]:
```suggestion
It's possible that someday that Sorbet could support a limited form of
```

**Pull request merged**